tests/external_module_dirs/external_modules/external_module/external_module.c

```diff
@@ -20,9 +20,10 @@
  */
 
 #include "external_module.h"
+#include "auto_init_priorities.h"
 #include "auto_init_utils.h"
 
-#define PRIO 1111
+#define PRIO AUTO_INIT_PRIORITY_AFTER(AUTO_INIT_PRIO_MOD_RANDOM)
 
```

Comment:
I suppose

```C
#define PRIO2 AUTO_INIT_PRIORITY_AFTER(PRIO)
```

would not work, right?

Reply:
The sorting is done by XFA alphabetically. The priorities just prefix numbers to the symbol name. So symbol `1337_foo` would always come after `1337_bar`, because `b` < `f`.

This may be good enough if both `foo` and `bar` really only care to be called after random is initialized and have no interdependencies.

Reply:
> #define PRIO2 AUTO_INIT_PRIORITY_AFTER(PRIO)

That’s also possible. Note the priority is 1042 instead of 1041. And note the alphabetic order if some modules have the same priority.

```diff
diff --git a/tests/external_module_dirs/external_modules/external_module/external_module.c b/tests/external_module_dirs/external_modules/external_module/external_module.c
index f520d66631..351fa55526 100644
--- a/tests/external_module_dirs/external_modules/external_module/external_module.c
+++ b/tests/external_module_dirs/external_modules/external_module/external_module.c
@@ -23,9 +23,16 @@
 #include "auto_init_priorities.h"
 #include "auto_init_utils.h"
 
+void bauto_init_external_module(void)
+{
+    external_module_initialized = true;
+}
+
 #define PRIO AUTO_INIT_PRIORITY_AFTER(AUTO_INIT_PRIO_MOD_RANDOM)
+#define PRIO2 AUTO_INIT_PRIORITY_AFTER(PRIO)
 
-AUTO_INIT(auto_init_external_module, PRIO);
+AUTO_INIT(bauto_init_external_module, PRIO2);
+AUTO_INIT(auto_init_external_module, PRIO2);
 
 bool external_module_initialized = false;
 char *external_module_message = "Linking worked";
```

`tests/external_module_dirs/bin/miot-nucleo-f767zi/tests_external_module_dirs.map`
```
 .roxfa.auto_init_xfa.5_1042
                0x0000000008003cc0       0x18 /home/fabian/forks/RIOT/tests/external_module_dirs/bin/miot-nucleo-f767zi/external_module/external_module.o
                0x0000000008003cc0                auto_init_xfa_auto_init_external_module
                0x0000000008003ccc                auto_init_xfa_bauto_init_external_module
```

`tests/external_module_dirs`
```
Type '/exit' to exit.
2022-10-13 17:48:25,548 # auto_init: ztimer_init (1010)
2022-10-13 17:48:25,551 # auto_init: auto_init_random (1040)
2022-10-13 17:48:25,555 # auto_init: auto_init_external_module (1042)
2022-10-13 17:48:25,559 # auto_init: bauto_init_external_module (1042)
2022-10-13 17:48:25,562 # auto_init: auto_init_miot_mtd (1249)
2022-10-13 17:48:25,566 # auto_init: auto_init_miot_node (1359)
2022-10-13 17:48:25,573 # main(): This is RIOT! (Version: 2022.07-devel-790-g44a6e-preprocessor_successor)
2022-10-13 17:48:25,575 # Message: Linking worked
2022-10-13 17:48:25,577 # Initialization worked!
2022-10-13 17:48:49,627 # Exiting Pyterm
```